app/app.go

```diff
@@ -181,6 +169,24 @@ func removeDatabase(cfg *config.Config) error {
 
 func openDB(cfg *config.Config) (database.Database, error) {
 	dbPath := databasePath(cfg)
+
+	doesVersionFileExist, err := checkDatabaseVersion(dbPath)
+	if err != nil {
+		return nil, err
+	}
+
 	log.Infof("Loading database from '%s'", dbPath)
-	return ldb.NewLevelDB(dbPath, leveldbCacheSizeMiB)
+	db, err := ldb.NewLevelDB(dbPath, leveldbCacheSizeMiB)
+	if err != nil {
+		return nil, err
+	}
+
+	if !doesVersionFileExist {
+		err := createDatabaseVersionFile(dbPath)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Is this also to avoid creating the directory?
as it could also crash between `ldb.NewLevelDB` and here, and have a db without a db version file